src/main/java/com/easypost/model/CarrierAccount.java

```diff
@@ -57,11 +57,16 @@ public CarrierAccount update(Map<String, Object> params, String apiKey) throws E
 		}
 
 		// delete
-		public void delete() throws EasyPostException {
-			this.delete(null);
+		public static CarrierAccount delete(String id) throws EasyPostException {
+			if(id.substring(0,2).equals("ca")){
+				return delete(id,null); 
+			}
+			else{
+				return null;
+			}
 		}
-		public void delete(String apiKey) throws EasyPostException {
-			request(RequestMethod.DELETE, instanceURL(CarrierAccount.class, this.getId()), null, CarrierAccount.class, apiKey);
+		public static CarrierAccount delete(String id, String apiKey) throws EasyPostException {
+			return request(RequestMethod.DELETE, instanceURL(CarrierAccount.class, id), null, CarrierAccount.class, apiKey);
```

Comment:
The CarrierAccount::Delete endpoint in EasyPost returns empty JSON, so I don't actually think there's anything here to return.